src/python/pants/backend/python/goals/export.py

```diff
@@ -252,7 +283,10 @@ async def do_export(
             f"--prompt={venv_prompt}",
             output_path,
         ]
-        if not export_subsys.options.py_hermetic_scripts:
+        if (
+            not export_subsys.options.py_hermetic_scripts
+            or req.resolve_name in export_subsys.options.py_non_hermetic_scripts_in_resolve
```

Comment:
```suggestion
            req.resolve_name in export_subsys.options.py_non_hermetic_scripts_in_resolve
            or not export_subsys.options.py_hermetic_scripts
```
Should we invert this so the new option takes precedence while the old option is deprecated?

Reply:
Yes, but this also probably need more nuanced logic to prevent a deprecation warning if the user has not changed `py_hermetic_scripts` from the default. I will have to see how deprecation warnings for options operate.

Reply:
Experimented a little on command line. Deprecation warning only shows up when I specify `--no-py-hermetic-scripts` which is what I would expect.

And pushed the swap of the options checks.